code/mobile/android/PhoneVR/app/src/main/cpp/alvr_main.cpp

```diff
@@ -107,13 +124,77 @@ AlvrFov getFov(CardboardEye eye) {
 
 AlvrPose getPose(uint64_t timestampNs) {
     AlvrPose pose = {};
+    bool returnLastPosition = false;
 
     float pos[3];
     float q[4];
     CardboardHeadTracker_getPose(CTX.headTracker, (int64_t) timestampNs, kLandscapeLeft, pos, q);
 
     auto inverseOrientation = AlvrQuat{q[0], q[1], q[2], q[3]};
     pose.orientation = inverseQuat(inverseOrientation);
+    if (!CTX.arcoreEnabled || (CTX.arcoreEnabled && !useARCoreOrientation)) {
+        CTX.lastOrientation = pose.orientation;
+    }
+
+    if (CTX.arcoreEnabled && CTX.arSession != nullptr) {
```

Comment:
this pattern kinda sucks; ideally we'd set `CTX.arcoreEnabled` to `false` if `arSession` fails to initialize